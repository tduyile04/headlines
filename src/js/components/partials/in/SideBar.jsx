import React from 'react';
import { withRouter } from 'react-router-dom';
import { SideNav, SideNavItem } from 'react-materialize';

/**
 * SideBar displaying the user profile and user saved
 * articles in the favourites
 * @param {any} user user object containing the profile data
 * @returns
 */
function SideBar({ user }) {
  let userObject;
  try {
    userObject = JSON.parse(user);
  } catch (e) {
    userObject = {
      image: '../../../../images/jared.jpg',
      name: 'anonymous',
      email: 'anonymous@unknown.com'
    };
  }
  return (
    <span className="grey darken-3 sidebtn">
      <SideNav
        trigger={
          <span
          className="grey darken-3">
            <i className="material-icons white-text sidebtn">menu</i>
          </span>
        }
        options={{ closeOnClick: true }}
        >
        <SideNavItem userView
          user={{
            background: '../../../../images/cloudsidebar.jpg',
            image: userObject.image,
            name: userObject.name,
            email: userObject.email
          }}
        />
        <SideNavItem subheader>Favourites</SideNavItem>
        <SideNavItem icon="cloud" href="/favourites">
          View Favourites
        </SideNavItem>
        <SideNavItem divider />
        <SideNavItem subheader>Recently Viewed Favourites</SideNavItem>
        <SideNavItem waves href="#!third">Still Random Post</SideNavItem>
      </SideNav>
    </span>
  );
}

export default withRouter(SideBar);
