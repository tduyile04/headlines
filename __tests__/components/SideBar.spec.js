import React from 'react';
import { shallow } from 'enzyme';
import SideBar from '../../src/js/components/partials/in/SideBar.jsx';

describe('<SideBar />', () => {
  it('should render 1 <SideBar />', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toHaveLength(1);
  });
  it('should display the same view for each render', () => {
    const wrapper = shallow(<SideBar />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return the expected user props', () => {
    const user = {
      name: 'jack doe',
      email: 'jackdoe@any.com',
      image: '/img/src/jack.jpg',
      idToken: '676tgybdy78h8e7ehue'
    };
    const wrapper = shallow(<SideBar user={user} />);
    expect(wrapper.instance().props.user).toBe(user);
  });
});
