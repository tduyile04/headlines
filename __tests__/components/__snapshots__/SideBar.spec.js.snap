// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should display the same view for each render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route
    render={[Function]}
/>,
  "nodes": Array [
    <Route
      render={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <withRouter(SideBar) />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Route
          render={[Function]}
/>,
        "_debugID": 4,
        "_renderedOutput": <Route
          render={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(SideBar) />,
}
`;
